### DISCUSSION TITLE
To know about digial art and images

### DISCUSSION THREAD

**Discussion by mdxabu** [KathiraveluLab/Beehive#3] [Q&A]
Dear @pradeeban ,

I hope this message finds you well. I am writing to inquire about the types of images and digital art that will be stored in our database. Could you share some samples of these images? This would greatly assist me in clarifying a point I am currently working on.

I am looking forward to your response.

Best regards,
F Mohamed Abdullah

**Comment by pradeeban** [comment]
Hi @mdxabu,
There will be some IRB, consent, and protocols involved before sharing the actual images, since these are drawn or taken by the individuals.

But for your testing purposes, you can use any photos you have taken or any drawing you have made (as in, your drawings scanned). There will be no difference as far as the technology is concerned (although your images are not going to give us the scientific value, since you are an individual, and not a group of people).

Does that answer your question?

Regards,
Pradeeban.

**Reply by mdxabu** [reply]
Can you clarify if the pictures and artwork are categorized by genre, such as healthcare or something else?

**Reply by pradeeban** [reply]
The persons submit the images (photos or scanned drawings). They are going into the Beehive's data store. It will be nice to be able to filter and view them by certain queries (for example, their condition, such as "unhoused," "person with AIDS," or "recovering alcoholic.")

So, probably, the images can be grouped by more than healthcare (since we include additional parameters as above beyond diseases).

Whether you index the images when you upload them or index them in batch - these are all implementation choices as this is a new project.

**Reply by mdxabu** [reply]
Okay, now I understand. Thanks for taking the time to clarify my doubts, @pradeeban

**Comment by mdxabu** [comment]
Dear @pradeeban ,

I believe I can create both the frontend and backend using Python and MongoDB. This approach will not only simplify implementation but also enable a broader range of operations.

Best regards.
@mdxabu

**Reply by pradeeban** [reply]
Sounds like a plan. Keep it simple. That way, you could implement easily, and then you can improve on it.

**Comment by nilaygit-10721** [comment]
dear @pradeeban 
To streamline the storage and management of images in the Beehive database, I propose implementing a robust system for indexing and categorization, supported by user-friendly frontend and scalable backend technology.

->Utilize pre-trained models (e.g., TensorFlow or PyTorch) to auto-tag images based on content, reducing manual effort.
->Enhance performance and enable future applications to query only required fields from the database, improving efficiency.

best regards,
nilay

**Reply by mdxabu** [reply]
I think I can answer your question. We have an idea to implement a tag-based image system. However, we can't use a pre-trained model to auto-tag the images because users upload images with their own emotions, so they only know the reasons/tags of the art. The pre-trained model should be wrong sometimes. But we can use it in the future.

@pradeeban, please let me know if I am correct.

**Reply by pradeeban** [reply]
@mdxabu Yes, you have perfectly articulated it!

**Reply by 01bps** [reply]
Hi @pradeeban ,
I have a small query regarding the tag-based image system discussed above, do you prefer users to manually tag images upon upload, or some auto-suggestions based on the metadata, allowing users to make the final decision?

**Reply by ishaanxgupta** [reply]
> Hi @pradeeban , I have a small query regarding the tag-based image system discussed above, do you prefer users to manually tag images upon upload, or some auto-suggestions based on the metadata, allowing users to make the final decision?

Yes, I also had the same doubt because some tagging about the image should be there.
Also what features can we build more on the user side?

**Reply by pradeeban** [reply]
The tags are manual. They are more of a narrative than a tag. Perhaps you are suggesting a brief "tag" in addition to the long textual narrative or a photovoice... That is ok. But they are manual and not automated by an algorithm, as @mdxabu mentioned.

The features should be centered around ease of use (remember the user base) and minimalism.

**Reply by 01bps** [reply]
Thank you for the clarification! 
Based on my understanding, I have an approach in mind that could help structure the tagging system while maintaining minimalism. I’d like to open a new discussion to explore this idea further.
Looking forward to your thoughts!

**Reply by ishaanxgupta** [reply]
> The tags are manual. They are more of a narrative than a tag. Perhaps you are suggesting a brief "tag" in addition to the long textual narrative or a photovoice... That is ok. But they are manual and not automated by an algorithm, as @mdxabu mentioned.
> 
> The features should be centered around ease of use (remember the user base) and minimalism.

Instead of a automated tag or category, can we try something with the emotions of the user which he wants to depict through the image and make the emotion as a tag?

**Reply by pradeeban** [reply]
Sure, that works.

**Comment by ishaanxgupta** [comment]
Also, what about a short audio message with the art?

**Reply by pradeeban** [reply]
That is great as well.
